triedb/pathdb/history_index_block.go

```diff
@@ -392,11 +380,14 @@ func (b *blockWriter) full() bool {
 //
 // This function is safe to be called multiple times.
 func (b *blockWriter) finish() []byte {
-	var buf []byte
-	for _, number := range b.restarts {
-		binary.BigEndian.PutUint16(b.scratch[:2], number)
-		buf = append(buf, b.scratch[:2]...)
+	restartsLen := len(b.restarts)
```

Comment:
The behavior is slightly different: 

in the original code, the trailing data is appended to the end of the existing b.data buffer, which likely has sufficient capacity to avoid reallocation. 

In the new code, the data copy is already required.

Reply:
I see, I need to have a try of checking the capacity of `b.data` before append b.restarts

Reply:
@rjl493456442 improved, PTAL